### PR TITLE
Fix peer dependencies

### LINKED ISSUE #373: Fix peer dependencies for TS 3
TypeScript 3 has been released, and has a small enough number of breaking changes that it shouldn't be too hard for us to officially support. I upgraded to 3.0.1 and it does seem to work well in CRA-TS, but we should probably verify that our dependencies have official TS 3 support.

```
npm WARN fork-ts-checker-webpack-plugin@0.2.10 requires a peer of typescript@^2.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-scripts-ts@2.16.0 requires a peer of typescript@2.x but none is installed. You must install peer dependencies yourself.
npm WARN ts-jest@22.0.1 requires a peer of typescript@2.x but none is installed. You must install peer dependencies yourself.
```

## Related
- [x] https://github.com/kulshekhar/ts-jest/issues/643
- [x] https://github.com/Realytics/fork-ts-checker-webpack-plugin/issues/133

### PR DESCRIPTION
**Pull request by nickmccurdy**
Fixes #373.

Continued from #376, merged with master, and tested to fix all peer dependency warnings.

### CODE REVIEW

**Comment by wmonk** [comment]
@nickmccurdy I messed up the deps on master, so if you could rebase then that should fix failing CI!

**Comment by nickmccurdy** [comment]
Most of the other errors seem to be false positives. This should be ready to merge. Sorry I couldn't figure out how to rebase these together into one commit without having to resolve the same conflicts again, could you squash merge (should be equivalent)?

**Comment by martijnthe** [comment]
Does this fix the warnings that `yarn` is giving me after creating a freshly generated react typescript app?

```
warning "workspace-aggregator-a8fb5286-b8b7-4413-ab03-0a73a4550814 > my-app > react-scripts-ts@2.17.0" has incorrect peer dependency "typescript@2.x".
warning "workspace-aggregator-a8fb5286-b8b7-4413-ab03-0a73a4550814 > my-app > react-scripts-ts > fork-ts-checker-webpack-plugin@0.2.10" has incorrect peer dependency "typescript@^2.1.0".
warning "workspace-aggregator-a8fb5286-b8b7-4413-ab03-0a73a4550814 > my-app > react-scripts-ts > ts-jest@22.0.1" has incorrect peer dependency "typescript@2.x".
```

**Comment by nickmccurdy** [comment]
Yes, it should.

**Comment by Fabianopb** [comment]
Nice, Jest 23 coming along. CI seems happy, any plans when this will be released? In 2.18.0 perhaps?

File: config/jest/typescriptTransform.js [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 'use strict';
 
-const tsJestPreprocessor = require('ts-jest/preprocessor');
+const tsJestPreprocessor = require('ts-jest');
```
Comment (r3nya) [inline config/jest/typescriptTransform.js]: Looks like we can remove this file and add
```json
"transform": {
   "^.+\\.tsx?$": "ts-jest",
}
```
into `scripts/utils/createJestConfig.js` config.

File: package.json [hunk 2]
```diff
@@ -10,26 +10,34 @@
...
+    "babel-runtime": "6.26.0",
     "case-sensitive-paths-webpack-plugin": "2.1.1",
     "chalk": "1.1.3",
     "css-loader": "0.28.7",
     "dotenv": "4.0.0",
     "dotenv-expand": "4.2.0",
     "extract-text-webpack-plugin": "3.0.2",
     "file-loader": "1.1.5",
-    "fork-ts-checker-webpack-plugin": "^0.2.8",
+    "fork-ts-checker-webpack-plugin": "0.4.4",
```
Comment (r3nya) [inline package.json]: The latest version is `0.4.9`.

File: config/jest/typescriptTransform.js (hunk 1 above)
Comment (DorianGrey) [reply config/jest/typescriptTransform.js]: Would be nice to be able to remove this workaround, but it should be tested thoroughly.
There were some issues in the past regarding the proper resolution in both ejected and non-ejected environments. This wrapper file was a (kinda hacky, but working) way to get around these issues.

File: package.json (hunk 2 above)
Comment (ohjames) [reply package.json]: On master this dependency version was upgraded and then downgraded back to 0.2.x, any idea why?

**Comment by nickmccurdy** [comment]
I'm closing this and moving back to CRA now that https://github.com/facebook/create-react-app/pull/4837 is merged.

**Pull request closed without merging**